Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererArrayBuilder.cs

```diff
@@ -164,7 +163,34 @@ private bool IsColumnVisible (BocColumnDefinition column)
 
       var columnAsDropDownMenuColumn = column as BocDropDownMenuColumnDefinition;
       if (columnAsDropDownMenuColumn != null)
-        return IsColumnVisibleForBocDropDownMenuColumnDefinition(columnAsDropDownMenuColumn);
+        if (!IsColumnVisibleForBocDropDownMenuColumnDefinition(columnAsDropDownMenuColumn))
+            return false;
+
+      var columnAsValidationErrorIndicatorColumn = column as BocValidationErrorIndicatorColumnDefinition;
+      if (columnAsValidationErrorIndicatorColumn != null)
+        return IsColumnVisibleForBocValidationErrorIndicatorColumnDefinition(columnAsValidationErrorIndicatorColumn);
+
+      return true;
+    }
+
+    private bool IsColumnVisibleForBocValidationErrorIndicatorColumnDefinition (BocValidationErrorIndicatorColumnDefinition columnAsValidationErrorIndicatorColumn)
+    {
+      if (columnAsValidationErrorIndicatorColumn.BehaviourWithoutValidationFailures == ErrorColumnWithoutValidationFailureBehaviour.Show)
+        return true;
+
+      var repository = ((IBocList?)columnAsValidationErrorIndicatorColumn.OwnerControl)?.ValidationFailureRepository;
+      if (repository == null)
+        return true;
+
+      if (columnAsValidationErrorIndicatorColumn.BehaviourWithoutValidationFailures == ErrorColumnWithoutValidationFailureBehaviour.HideUnlessAnyValidationFailures)
+      {
+        return repository.GetListFailureCount() + repository.GetRowAndCellFailureCount() != 0;
+      }
+
+      if (columnAsValidationErrorIndicatorColumn.BehaviourWithoutValidationFailures == ErrorColumnWithoutValidationFailureBehaviour.HideUnlessRowOrCellValidationFailures)
+      {
+        return repository.GetRowAndCellFailureCount() != 0;
+      }
```

Comment:
I would ammend @a-ctor comment on the switch statement with:
1. move the guard condition  (repostiroy == null) to the top. 
2. make the remainding three cases into an if/else or a switch.

Note: to be very precise, having no OwnerControl would actually be a Assertion. This should never happen. I think we should add this guard (`Assertion.IsNotNull(OwnerControl)`). Disclaimer: I don't know how the other places usually handle OwnerControl==null. We should be reasonably cosinstent.